File: src/test/java/pages/base/HomePage.java
package pages.base;

import com.microsoft.playwright.Page;

public class HomePage {
    private final Page page;
    private final String profileMenu = "//span[text()='Profiles']";
    private final String addNewProfile = "//button[text()=' Add New Profile ']";
    private final String manufacturerProfile = "//h5[text()='Manufacturer']";
    private final String retailer = "//button[text()=' Select Retailer ']";
    private final String Circana = "//button[text()=' Select Circana Profile ']";
    private final String geography = "//span[text()='Geography']";

    public HomePage(Page page){
        this.page = page;
    }

    public String getTitle(){
        return page.title();
    }

    public void goToProfiles(){
        page.click(profileMenu);
    }

    public void goToManufacturer(){
        page.click(profileMenu);
        page.click(addNewProfile);
        page.click(manufacturerProfile);
    }

    public void goToRetailer(){
        page.click(profileMenu);
        page.click(addNewProfile);
        page.click(retailer);
    }

    public void goToCircana(){
        page.click(profileMenu);
        page.click(addNewProfile);
        page.click(Circana);
    }

    public void goToGeography(){
        page.click(geography);
    }
}

File: pages/homePage.ts
import { Page, Locator } from '@playwright/test';
import envConfig from '../config/env';

export class HomePage {
  readonly page: Page;
  readonly profileMenu: Locator;
  readonly addNewProfileButton: Locator;
  readonly manufacturerOption: Locator;
  readonly retailerOption: Locator;
  readonly circanaOption: Locator;
  readonly geographyLink: Locator;

  constructor(page: Page) {
    this.page = page;
    this.profileMenu = page.getByRole('link', { name: /profiles/i });
    this.addNewProfileButton = page.getByRole('button', { name: /add new profile/i });
    this.manufacturerOption = page.getByRole('button', { name: /select manufacturer/i });
    this.retailerOption = page.getByRole('button', { name: /select retailer/i });
    this.circanaOption = page.getByRole('button', { name: /select circana profile/i });
    this.geographyLink = page.getByRole('link', { name: /geography/i });
  }

  async goto(): Promise<void> {
    await this.page.goto('/');

    await this.navigateToApplication();
  }

  async navigateToApplication() {
    switch (envConfig.name) {

      case 'dev':
        await this.page.locator('#menu-left').click();
        await this.page.getByText('#menu-Geography-Builder').click();
        break;

      case 'qa':
        await this.page.locator('#menu-left').click();
        await this.page.getByText('#menu-Geography-Builder(qa)').click();
        break;
        break;
    }
  }

  async title(): Promise<string> {
    return this.page.title();
  }

  async openProfiles(): Promise<void> {
    await this.profileMenu.click();
  }

  async openManufacturers(): Promise<void> {
    await this.openProfiles();
    await this.addNewProfileButton.click();
    await this.manufacturerOption.click();
  }

  async openRetailer(): Promise<void> {
    await this.openProfiles();
    await this.addNewProfileButton.click();
    await this.retailerOption.click();
  }

  async openCircana(): Promise<void> {
    await this.openProfiles();
    await this.addNewProfileButton.click();
    await this.circanaOption.click();
  }

  async openGeography(): Promise<void> {
    await this.geographyLink.click();
  }
}
